### ISSUE TITLE
Crash when clicking entry (branch main)

### ISSUE THREAD

**Issue by svmorris** [Authentura/codectrl#30]
The main window crashes with a "permission denied" error when trying to select one of the log entries. This bug happens on both mine and Dasiths computer. 

Some other information:
- Tested with nodejs logger
- Latest version of codectrl on main branch
- Latest rust nightly (just installed it)

Haven't had time to test with others, but no logger should be able to crash the main thread.

Error:
```
Server has started on 0.0.0.0:3001...
Failed to create /builder/.cache for shader cache (Permission denied)---disabling.
thread 'main' panicked at 'already immutably borrowed', /home/antone/.cargo/registry/src/github.com-1ecc6299db9ec823/atomic_refcell-0.1.8/src/lib.rs:151:23
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
https://user-images.githubusercontent.com/46561460/161345723-8655cb3b-6c1c-458c-9661-7dfc7199cad7.mp4

**Comment by STBoyden** [comment]
I'm having the same issue, but it's the "the already immutably borrowed" part that is the issue. Trying to figure out a fix currently.

**Comment by svmorris** [comment]
oh ok, cheers!

**Comment by STBoyden** [comment]
Tracked in the egui-0.17-crash branch

**Comment by STBoyden** [comment]
I have managed to narrow it down to _one_ line, but it still doesn't seem to be an obvious cause or fix. 

[Offending line](https://github.com/Authentura/codectrl/blob/262aa6962b6babc281f43bf610a995511fbcc42a/codectrl-gui/src/components/details_view_components/information_grid.rs#L19)

**Comment by STBoyden** [comment]
This _should_ be fixed from the 043b007. Any further issues please re-open this issue
